src/server/api/routers/auth.ts

```diff
@@ -85,4 +85,57 @@ export const loginRouter = createTRPCRouter({
         success: true,
       };
     }),
+
+  create: publicProcedure
+    .input(z.object({ email: z.string(), password: z.string() }))
+    .mutation(async ({ input }) => {
+      try {
+        const user = await db.query.users.findFirst({
+          where: eq(users.email, input.email),
+        });
+
+        if (user) {
+          throw new TRPCError({
+            code: "CONFLICT",
+            message: "User already exists",
+          });
+        }
+
+        const salt: string = await bcrypt.genSalt(10);
+        const hashedPassword: string = await bcrypt.hash(input.password, salt);
+
+        const adapter = authOptions.adapter;
+
+        if (adapter?.createUser === undefined) {
+          throw new TRPCError({
+            code: "INTERNAL_SERVER_ERROR",
+            message: "Adapter not found",
+          });
+        }
+
+        // TODO: verify email
+        const createdUser = await adapter.createUser({
+          email: input.email,
+          emailVerified: new Date(),
+        });
+
+        await db
+          .update(users)
+          .set({
+            password: hashedPassword,
+          })
+          .where(eq(users.id, createdUser.id));
+        return {
+          success: true,
+        };
+      } catch (error) {
+        throw error instanceof TRPCError
+          ? error
+          : new TRPCError({
+              code: "INTERNAL_SERVER_ERROR",
+              message:
+                "Failed to create user with password" + JSON.stringify(error),
+            });
+      }
+    }),
```

Comment:
I thought this was already merged in?? Might be showing here because you haven't merged with latest? I'm not sure...